networkx/algorithms/centrality/tests/test_domirank.py

```diff
@@ -0,0 +1,197 @@
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+    def test_P3(self):
+        """DomiRank centrality: P3"""
+        G = nx.path_graph(3)
+        b, _, __ = nx.domirank(G)
+        b_answer = {
+            0: -2.3477606773376465,
+            1: 4.496389865875244,
+            2: -2.3477606773376465,
+        }
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+        b_answer = {0: -2.366595958759845, 1: 4.523026818398872, 2: -2.366595958759845}
+        b, _, __ = nx.domirank(G, analytical=True)
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+        b_answer = {
+            0: 0.13808214664459229,
+            1: 0.609459638595581,
+            2: 0.13808214664459229,
+        }
+        b, _, __ = nx.domirank(G, sigma=0.5)
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+    def test_P3_dt(self):
+        """DomiRank centrality: P3"""
+        G = nx.path_graph(3)
+        b_answer = {
+            0: -2.3666865825653076,
+            1: 4.523157596588135,
+            2: -2.3666865825653076,
+        }
+        b, _, __ = nx.domirank(G, dt=0.5)
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+    def test_P3_eps(self):
+        """DomiRank centrality: P3"""
+        G = nx.path_graph(3)
+        b_answer = {
+            0: -2.3477983474731445,
+            1: 4.496443748474121,
+            2: -2.3477983474731445,
+        }
+        b, _, bBool = nx.domirank(G, epsilon=1e-7)
+        assert bBool == True
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+    def test_P3_max_iter(self):
+        """DomiRank centrality: P3"""
+        G = nx.path_graph(3)
+        b_answer = {0: -2.364804744720459, 1: 4.5204758644104, 2: -2.364804744720459}
+        b, _, bBool = nx.domirank(G, max_iter=5000)
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+    def test_P3_check_step(self):
+        """DomiRank centrality: P3"""
+        G = nx.path_graph(3)
+        b_answer = {0: -2.346951484680176, 1: 4.495236396789551, 2: -2.346951484680176}
+        b, _, bBool = nx.domirank(G, check_step=1)
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+
+class TestDomirankDirected:
+    def __init__(self):
+        self.G = nx.DiGraph()
+
+        edges = [
+            (1, 2),
+            (1, 3),
+            (2, 4),
+            (3, 2),
+            (3, 5),
+            (4, 2),
+            (4, 5),
+            (4, 6),
+            (5, 6),
+            (5, 7),
+            (5, 8),
+            (6, 8),
+            (7, 1),
+            (7, 5),
+            (7, 8),
+            (8, 6),
+            (8, 7),
+        ]
+
+        self.G.add_edges_from(edges)
+        self.G = self.G.reverse()
+        self.evc = {
```

Comment:
I'm assuming the variable names are copy-paste oversights. You should make them fit the fuction you're actually testing, e.g. with `self.drc` and `self.drca`.

Reply:
Thank you, great catch!